### PR TITLE
Codechange: Use constants for tree growth stages and tile update frequency

### PR DESCRIPTION
**Pull request by ldpl**
## Motivation / Problem

Doing all the unrelated refactoring in a separate PR so it can be discussed and merged without having to constantly reshuffle around #11955 that is complex enough as it is.

## Description

Just all the requested refactoring that is not directly related to the change.

## Checklist for review

Some things are not automated, and forgotten often. This list is a reminder for the reviewers.
* The bug fix is important enough to be backported? (label: 'backport requested')
* This PR touches english.txt or translations? Check the [guidelines](https://github.com/OpenTTD/OpenTTD/blob/master/docs/eints.md)
* This PR affects the save game format? (label 'savegame upgrade')
* This PR affects the GS/AI API? (label 'needs review: Script API')
    * ai_changelog.hpp, game_changelog.hpp need updating.
    * The compatibility wrappers (compat_*.nut) need updating.
* This PR affects the NewGRF API? (label 'needs review: NewGRF')
    * newgrf_debug_data.h may need updating.
    * [PR must be added to API tracker](https://wiki.openttd.org/en/Development/NewGRF/Specification%20Status)

### CODE REVIEW

File: src/tree_cmd.cpp [hunk 1]
```diff
@@ -750,15 +751,15 @@ static void TileLoop_Trees(TileIndex tile)
 
 						TreeType treetype = GetTreeType(tile);
 
-						tile += TileOffsByDir((Direction)(Random() & 7));
+						tile += TileOffsByDir(static_cast<Direction>(Random() & 7));
```
Comment (2TallTyler) [inline src/tree_cmd.cpp]:
```suggestion
						tile += TileOffsByDir(static_cast<Direction>(Random() & DIR_END));
```

**Review by 2TallTyler — COMMENTED** [review]
Thanks for tackling this tech debt. I know it's not the fun part of development. 🙂

**Review by 2TallTyler — COMMENTED** [review]
How about a new commit title: `Codechange: Use constants for tree growth stages and tile update frequency`? 🙂

**Comment by ldpl** [comment]
... and some static casts for extra lolz
But, sure, whatever

**Review by 2TallTyler — DISMISSED** [review]

File: src/landscape.cpp [hunk 2]
```diff
@@ -82,6 +82,9 @@ extern const uint8_t _slope_to_sprite_offset[32] = {
 	0, 0, 0, 0, 0, 0, 0, 16, 0, 0,  0, 17,  0, 15, 18, 0,
 };
 
+static const uint TILE_UPDATE_FREQUENCY_LOG = 8;  // The logarithm of how many ticks it takes between tile updates (log base 2).
+static const uint TILE_UPDATE_FREQUENCY = 1 << TILE_UPDATE_FREQUENCY_LOG;  // How many ticks it takes between tile updates (has to be a power of 2).
```
Comment (PeterN) [inline src/landscape.cpp]: Should be `///< ` for doxygen comments.

**Review by 2TallTyler — DISMISSED** [review]

File: src/tree_map.h [hunk 3]
```diff
@@ -57,6 +57,19 @@ enum TreeGround {
...
+ * Enumeration for tree growth stages.
+ *
+ * This enumeration defines the stages of tree growth for tiles with trees on it.
+ */
+enum class TreeGrowthStage : uint {
+	Growing1 = 0,  ///< First stage of growth
+	Growing2 = 1,  ///< Second stage of growth
+	Growing3 = 2,  ///< Third stage of growth
+	Grown = 3,  ///< Fully grown tree
+	Dying1 = 5,  ///< First stage of dying
```
Comment (nielsmh) [inline src/tree_map.h]: Does the integral value 4 have any meaning in this?

Comment (ldpl) [reply src/tree_map.h]: Oops :) For some reason I thought there are only 2 dying stages

Comment (ldpl) [reply src/tree_map.h]: Actually, the last one is kinda fully dead so might as well name it as such I guess

**Review by rubidium42 — APPROVED** [review]

**Pull request merged**